src/resources/bootstrap.cpp

```diff
@@ -0,0 +1,491 @@
+            continue;
+        } else {
+            log(LOG_SEVERITY_INFO, "Found suitable bbf for bootstraping: %s", bbf.get());
+            bbf_found = true;
+            break;
+        }
+    }
+
+    if (!bbf_found) {
+        log(LOG_SEVERITY_WARNING, "Failed to find a .bbf file");
+        return false;
+    }
+
+    return true;
+}
+
+static bool do_bootstrap(buddy::resources::ProgressHook progress_hook) {
+    BootstrapProgressReporter reporter(progress_hook);
+    Path source_path("/");
+
+    reporter.report("Looking for .bbf");
+    if (!find_suitable_bbf_file(source_path)) {
+        return false;
+    }
+
+    reporter.report("Preparing bootstrap");
+    // clear installed resources
+    buddy::resources::InstalledRevision::clear();
+    remove("/internal/res");
+
+    // open the bbf
+    std::unique_ptr<FILE, FILEDeleter> bbf(fopen(source_path.get(), "rb"));
+    if (bbf.get() == nullptr) {
+        log(LOG_SEVERITY_WARNING, "Failed to open %s", source_path.get());
+        return false;
+    }
+
+    // mount the filesystem stored in the bbf
+    ScopedFileSystemLittlefsBBF scoped_bbf_mount(bbf.get());
+    if (!scoped_bbf_mount.mounted()) {
+        log(LOG_SEVERITY_INFO, "BBF mounting failed");
+        return false;
+    }
+
+    // calculate stats for better progress reporting
+    ResourcesScanResult scan_result;
+    source_path.set("/bbf");
+    if (!scan_resources_folder(source_path, scan_result)) {
+        log(LOG_SEVERITY_ERROR, "Failed to scan the /bbf directory");
+    }
+    log(LOG_SEVERITY_INFO, "To copy: %i files, %i directories, %i bytes",
+        scan_result.files_count, scan_result.directories_count, scan_result.total_size_of_files);
+    reporter.assign_scan_result(scan_result);
+
+    // open the bbf's root dir
+    std::unique_ptr<DIR, DIRDeleter> dir(opendir("/bbf"));
+    if (dir.get() == nullptr) {
+        log(LOG_SEVERITY_WARNING, "Failed to open /bbf directory");
+        return false;
+    }
+
+    // copy the resources
+    reporter.report("Copying files");
+    source_path.set("/bbf");
+    Path target_path("/internal/res");
+    if (!copy_resources_directory(source_path, target_path, reporter)) {
+        log(LOG_SEVERITY_ERROR, "Failed to copy resources");
+        return false;
+    }
+
+    // save the installed revision
+    if (!buddy::resources::InstalledRevision::set(buddy::resources::revision::required_revision)) {
+        log(LOG_SEVERITY_ERROR, "Failed to save installed resources revision");
+        return false;
+    }
+
+    return true;
+}
+
+bool buddy::resources::bootstrap(ProgressHook progress_hook) {
+    while (true) {
+        bool success = do_bootstrap(progress_hook);
+        if (success) {
+            log(LOG_SEVERITY_INFO, "Bootstrap successful");
+            return true;
+        } else {
+            log(LOG_SEVERITY_INFO, "Bootstrap failed. Retrying in a moment...");
+            osDelay(1000);
```

Comment:
So, if I don't have the right file around, will this just wait during boot (without much info on the screen) doing nothing?

Reply:
Yep, rn, it is just gonna be saying „Waiting for bbf“. Until release, we definitely need to improve the UX in such cases. But for now, I think it is enough